### PR TITLE
Add some basic testing for the Material class

### PR DESCRIPTION
**Pull request by psavery**
This is just adding a little framework for testing the Material class,
through which we can add more tests.

For: #214

### CODE REVIEW

File: tests/test_material.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+def are_close(v1, v2, tol=FLOAT_TOL):
+    return abs(normalize_unit(v1) - normalize_unit(v2)) < tol
+
+
+def lparms_are_close(lparms, indices, tol=FLOAT_TOL):
+    first = lparms[indices[0]]
+    return all(are_close(first, lparms[i], tol) for i in indices[1:])
+
+
+def test_sgnum_setter(default_material):
```
Comment (psavery) [inline tests/test_material.py:40]: Just to explain what is going on here, pytest does a few things automatically based upon how you name things.

If you start a function name with `test_`, pytest assumes it is a test and tries to run it.

Arguments that you put in the `test_` function can be pytest fixtures. In this case, the argument is named `default_material`, which, via name matching, pytest automatically associates with [the fixture function defined above named `default_material`](https://github.com/HEXRD/hexrd/pull/220/files#diff-428844a4010d0b1e799aacd172b9ac423226cd6ce830c4c75858b434be6df8b5R13-R15). It automatically calls that function and passes the returned value as the argument to this test function.

File: tests/test_material.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    assert are_close(lparms[5], 90)
+
+    default_material.sgnum = 165
+    lparms = default_material.latticeParameters
+
+    # Gamma should be 120, the other angles should be 90
+    assert not lparms_are_close(lparms, [3, 4, 5])
+    assert lparms_are_close(lparms, [3, 4])
+    assert are_close(lparms[3], 90)
+    assert are_close(lparms[5], 120)
```
Comment (psavery) [inline tests/test_material.py:57]: So, this is what is actually being tested here right now. You can see it being tested [here](https://github.com/HEXRD/hexrd/runs/2122961429?check_suite_focus=true#step:9:201).

File: tests/test_material.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        # Check that the values for ruby match
+        ruby = materials['ruby']
+        params = f['ruby']['LatticeParameters'][()]
+
+        # Convert to angstroms...
+        for i in range(3):
+            params[i] *= 10
+
+        for i in range(6):
+            assert are_close(params[i], ruby.latticeParameters[i])
```
Comment (psavery) [inline tests/test_material.py:83]: And here is loading a materials file and checking that the lattice parameters for ruby match.

**Comment by psavery** [comment]
If you want to run the tests locally, you just need to make sure you have also cloned the [examples repo](https://github.com/hexrd/examples), and set this environment variable to the path on your file system:
```bash
export HEXRD_EXAMPLE_REPO_PATH=/path/to/examples
```

Then, in the hexrd repo, you can pip install the requirements and run the tests like
```python
pip install -r tests/requirements-dev.txt
pytest -s tests
```

Or, specially only run the materials test:
```python
pytest -s tests/test_material.py
```

That will skip the fit grains and find orientations tests, and only run the materials tests.

**Comment by joelvbernier** [comment]
I'll add a bit to this then merge.

**Comment by joelvbernier** [comment]
I need to pass along some commits for this... or should we merge this and add additional testing in a later commit?

**Comment by psavery** [comment]
@joelvbernier I'm good with merging this and adding extra testing later if that's good with you.

**Review by saransh13 — APPROVED** [review]
I agree. This is a good start. We can add more tests later.
LGTM.

**Pull request merged**